Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs

```diff
@@ -76,18 +98,19 @@ private void OnRoundEndText(RoundEndTextAppendEvent ev)
         ev.AddLine(Loc.GetString("nukeops-list-start"));
         foreach (var nukeop in _aliveNukeops)
         {
-            ev.AddLine($"- {nukeop.Key.Session?.Name}");
+            ev.AddLine($"- {nukeop.Key?.Session?.Name}");
         }
     }
 
-    private void OnMobStateChanged(MobStateChangedEvent ev)
+    private void OnMobStateChanged(EntityUid uid, NukeOperativeComponent component, MobStateChangedEvent ev)
     {
         if (!RuleAdded)
             return;
 
-        if (!_aliveNukeops.TryFirstOrNull(x => x.Key.OwnedEntity == ev.Entity, out var op)) return;
+        if (!_aliveNukeops.TryFirstOrNull(x => x.Key?.OwnedEntity == ev.Entity, out var op))
+            return;
```

Comment:
Done. I've replaced it with an entity query. In theory, other entities can have the NukeOperativeComponent added by admins and they will be counted as being on the nuke operative team.

Some questions:
- Should the round end if there are nuke op ghost roles still available?
- If other entities did get the NukeOperativeComponent added, should they be part of the killed operative check? I'm considering adding a bool datafield to indicate this.

Reply:
for your first question, i feel like it should if the ghost role operatives aren't able to reasonably get to the station (i.e., their shuttle is not on the moon base map, and they're stuck on the nukie base as a result if all other operatives are dead)

for consideration, though: leaving the round running *may* come with a chance that the crew would attempt to invade the nuclear operative base through the operative shuttle *if* they have control over it, meaning that the operatives can try to take back their ship and continue the mission, but that's all dependent on the players and it may drag out a round for too long

Reply:
For now, I'll have it check for nuke operatives being alive on the same map that the shuttle is present in. This does introduce a problem, where the round will end if the shuttle is hijacked and taken to a map that isn't the outpost or the station and the check is triggered. 

`CheckRoundShouldEnd` should also check if operatives are present on the target station. I'm not sure if I should leave that to be implemented in #10220 , or take some code from that PR and implement the logic in this branch?

Reply:
i can implement this behavior in #10220 once this is merged in, since the scope of that focuses all on nuclear operative win conditions